-
  In order to test marketplace.announcement I create a new announcement
-
  !context
    uid: 'base_community.user_luc'
-
  !record {model: marketplace.announcement, id: announcement_test_0}:
    name: I'm offering services about Odoo Customization
    type: offer
    description: This include python development, formation, helpdesk, but not frontend development on Odoo CMS
    category_id: marketplace.category_odoo
    tag_ids:
      - marketplace.tag_odoo_python
      - marketplace.tag_odoo_formation
      - marketplace.tag_odoo_helpdesk
    partner_id: base.res_partner_12
    infinite_qty: 1
    uom_id: product.product_uom_hour
    currency_ids:
      - model: "marketplace.announcement"
        field: "currency_ids"
        price_unit: 50
        currency_id: account_wallet.COM
-
  I check that nitially the announcement is in the "Draft" state
-
  !assert {model: marketplace.announcement, id: announcement_test_0}:
    - state == 'draft'
-
  I check that mark can't publish it
-
  !context
    uid: 'base_community.user_mark'
-
  !python {model: marketplace.announcement}: |
    from openerp.osv import orm, osv
    from openerp import workflow
    try:
      workflow.trg_validate(uid, 'marketplace.announcement', ref('announcement_test_0'), 'announcement_draft_open', cr)
      assert False, "Mark was able to publish the announcement!"
    except osv.except_osv:
      pass
-
  I check that Luc can publish it
-
  !context
    uid: 'base_community.user_luc'
-
  !python {model: marketplace.announcement}: |
    from openerp import workflow
    workflow.trg_validate(uid, 'marketplace.announcement', ref('announcement_test_0'), 'announcement_draft_open', cr)
-
  !assert {model: marketplace.announcement, id: announcement_test_0, string: "Luc wasn't able to publish the announcement!"}:
    - state == 'open'
-
  I check that Mark can't close it
-
  !context
    uid: 'base_community.user_mark'
-
  !python {model: marketplace.announcement}: |
    from openerp.osv import orm, osv
    from openerp import workflow
    try:
      workflow.trg_validate(uid, 'marketplace.announcement', ref('announcement_test_0'), 'announcement_open_done', cr)
      assert False, "Mark was able to close the announcement!"
    except osv.except_osv:
      pass
-
  I check that Luc can close it
-
  !context
    uid: 'base_community.user_luc'
-
  !python {model: marketplace.announcement}: |
    from openerp import workflow
    workflow.trg_validate(uid, 'marketplace.announcement', ref('announcement_test_0'), 'announcement_open_done', cr)
-
  !assert {model: marketplace.announcement, id: announcement_test_0, string: "Luc wasn't able to close the announcement!"}:
    - state == 'done'
-
  I check that Luc can reopen it
-
  !python {model: marketplace.announcement}: |
    self.reset_workflow(cr, uid, [ref('announcement_test_0')])
-
  !assert {model: marketplace.announcement, id: announcement_test_0, string: "Luc wasn't able to re-open the announcement!"}:
    - state == 'open'